### ISSUE TITLE
HeroFlow: emptying cart means hero flow is not shown

### ISSUE THREAD

**Issue by autumnfjeld** [Automattic/wp-calypso#57358]
This is lower priority as it isn't a common use case, probably not blocking shipping, but should be fixed.

1. go through signup and pick paid plan
2. when in checkout , remove everything from the cart
3. see that user is routed to https://wordpress.com/plans/auttest26boct.wordpress.com

Should user be routed to the intent screen?

**Comment by arthur791004** [comment]
I've discussed with @ollierozdarz here: p1630576003088500-slack-C029SB8JT8S and we keep them on the plan page to avoid losing the chance for them to select another plan. But I forgot this one...and I'm not sure it's possible or not
> Once either checkout is complete, or they choose Free (and have no paid domain), then we should show them the design picker

**Comment by roo2** [comment]
Some takeaways from the thread:
We want to give the user another chance to choose a paid plan and domain:
Ollie Rozdarz: p1630580729094500/1630576003.088500-slack-C029SB8JT8S
> It doesn’t make sense to me to then continue with onboarding them since we lose the opportunity for them to re-select a plan


Ollie assumed that the user would be taken back to the `/signup/plans` page in the signup flow. I'm actually not sure why we don't do this? 

Would it be neccessary to make checkout an onboarding flow step in order to implement this? p1630630435146300/1630576003.088500-slack-C029SB8JT8S. 

We wouldn't want to take the user to the`/start/plans` step if they've removed a paid domain from their cart. Then they would only have options to purchase the same domain that was just removed, and they wouldn't see the free plan option.

I will look into whether it's possible to redirect the user that empties their cart to `/start/plans` if they chose a free domain and to `/start/domains` if they removed a paid domain. Or event just `/start/domains` for all cases.
  
Other ideas from the thread:
 * Checkout as an onboarding flow step p1630630435146300/1630576003.088500-slack-C029SB8JT8S
  * For users who didn't use the design picker, add design picker to My Home checklist, If they navigate to the editor and still haven’t been shown the design picker, show it then p1630618439116800/1630576003.088500-slack-C029SB8JT8S.
 * Take the Plans page with the pulsating cart button and turn it into a first optional step in the new post-signup onboarding. p1630581785104800/1630576003.088500-slack-C029SB8JT8S

**Comment by roo2** [comment]
I had another discussion in slack p1635900925442100-slack-C029SB8JT8S summary:
If we redirected to the `/start/domains` page, we have to make sure that [the steps we took to preventing creating duplicate ](https://github.com/Automattic/wp-calypso/pull/45040) sites are right for this use case.

Because this would be a change that might impact purchasing rates, we should discuss it in a P2 first.

Since redirecting to the calypso `/plans` page is the existing behavior, I don't think that we should necessarily see this issue as part of the hero-flow work, and rather, as something that we should come back to after some more discussion.

What do you think @autumnfjeld?

**Comment by autumnfjeld** [comment]
This sounds like something bigger than just Blogger flow related. So let's pause this effort and watch if this issue is brought up elsewhere.   And I think your idea to discuss this via P2 for a wider audience after we launch the hero flow is a great idea.

Thanks for investigating @eru!

**Comment by autumnfjeld** [comment]
I'm going to close this due to all work going on with chaning paid plans, refactoring, and new onboarding flows.
